great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -0,0 +1,496 @@
+
+
+            evaluation_result = func(self, column, *args, **kwargs)
+
+            if ('success' not in evaluation_result) or \
+                ('result_obj' not in evaluation_result) or \
+                ('observed_value' not in evaluation_result['result_obj']):
+                raise ValueError("Column aggregate expectation failed to return required return information.")
+
+            return_obj = {
+                'success': bool(evaluation_result['success'])
+            }
+
+            if result_format['result_obj_format'] == 'BOOLEAN_ONLY':
+                return return_obj
+
+            count_query = select([
+                sa_func.count().label('element_count'),
+                sa_func.sum(
+                    case([(sa_column(column) == None, 1)], else_=0)
+                ).label('null_count'),
+            ]).select_from(table(self.table_name))
+
+            count_results = self.engine.execute(count_query).fetchone()
+
+            return_obj['result_obj'] = {
+                'observed_value': evaluation_result['result_obj']['observed_value'],
+                "element_count": count_results['element_count'],
+                "missing_count": count_results['null_count'],
+                "missing_percent": count_results['null_count'] * 1.0 / count_results['element_count'] if count_results['element_count'] > 0 else None
+            }
+
+            if result_format['result_obj_format'] == 'BASIC':
+                return return_obj
+
+            if 'details' in evaluation_result['result_obj']:
+                return_obj['result_obj']['details'] = evaluation_result['result_obj']['details']
+
+            if result_format['result_obj_format'] in ["SUMMARY", "COMPLETE"]:
+                return return_obj
+
+            raise ValueError("Unknown result_format %s." % (result_format['result_obj_format'],))
+
+        return inner_wrapper
+
+class SqlAlchemyDataSet(MetaSqlAlchemyDataSet):
+
+    # def __init__(self, connection_string, table_name):
+    def __init__(self, engine, table_name):
+        super(SqlAlchemyDataSet, self).__init__()
+        # We are intentionally not adding default expectations here, thinking about the future of non-tabular datasets
+        self.table_name = table_name
+        # self.engine = create_engine(connection_string)
+        self.engine = engine
+
+        insp = reflection.Inspector.from_engine(engine)
+        self.columns = insp.get_columns(self.table_name)
+
+    def add_default_expectations(self):
+        """
+        The default behavior for PandasDataSet is to explicitly include expectations that every column present upon initialization exists.
+        """
+        for col in self.columns:
+            self.append_expectation({
+                "expectation_type": "expect_column_to_exist",
+                "kwargs": {
+                    "column": col["name"]
+                }
+            })
+
+    def _is_numeric_column(self, column):
```

Comment:
?

Reply:
hm. We've brought typing in via the backdoor.

Reply:
True. I'd argue it's no more severe than what happens right now with the implied type coercions in python (that we inherit in PandasDataSet). Actually, this was necessary to ensure we're staying closer between the two right now. But I'm definitely looking forward to a more elegant solution, e.g. as implied in #207 .

Reply:
:+1: